crates/sui-json-rpc/src/name_service.rs

```diff
@@ -224,3 +232,59 @@ pub struct NameRecord {
     /// Additional data which may be stored in a record
     pub data: VecMap<String, String>,
 }
+
+impl NameRecord {
+    /// Leaf records expire when their parent expires.
+    pub fn is_leaf_record(&self) -> bool {
+        self.expiration_timestamp_ms == LEAF_EXPIRATION_TIMESTAMP
```

Comment:
does this mean that the NameRecord's expiration_timestamp_ms self-counts down? Or is that something we'd have to calculate every time we load a record?

Reply:
Oh or does it mean that we always set expiration to 0, and we need to look at its parent

Reply:
So the expiration timestamp is a UTC timestamp with ms for the future (based off clock). 
For leaf names, we've set this to 0 on-chain, as their expiration is derived by the parent's expiration!